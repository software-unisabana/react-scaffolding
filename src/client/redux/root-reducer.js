import { combineReducers } from 'redux';

import itemReducer from './item/item.reducer';

const rootReducer = combineReducers({
	products: itemReducer,
});

export default rootReducer;
