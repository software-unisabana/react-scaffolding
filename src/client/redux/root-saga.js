import { all, call } from 'redux-saga/effects';

import { itemSagas } from './item/item.sagas';

export default function* rootSaga() {
	yield all([call(itemSagas)]);
}
