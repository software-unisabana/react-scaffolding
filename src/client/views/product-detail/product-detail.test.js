import React from 'react';
import { mount } from 'enzyme';
import { combineReducers, createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import ProductDetailView from './product-detail.component.jsx';

export const createMockStore = ({ state, reducers }) => {
	const store = createStore(combineReducers(reducers), state);
	return {
		...store,
	};
};

describe('ProductDetailView', () => {
	let wrapper;
	let store;

	beforeEach(() => {
		const mockReducer = (
			state = {
				isFetching: true,
			},
			action
		) => state;

		const mockState = {
			products: {
				isFetching: true,
			},
		};

		store = createMockStore({
			state: mockState,
			reducers: { products: mockReducer },
		});
		const mockProps = {
			match: {
				params: {
					id: 'MLA689389817',
				},
			},
			fetchItemDetailStart: jest.fn(),
			itemDetail: {
				id: 'MLA689325634',
				title: 'Ps5',
				price: {
					currency: 'ARS',
					amount: 38500,
					decimals: 2,
				},
				picture: '',
				condition: 'new',
				free_shipping: false,
				sold_quantity: 5,
				description: 'ps%',
			},
			isLoading: true,
		};

		wrapper = mount(
			<BrowserRouter>
				<Provider store={store}>
					<ProductDetailView {...mockProps} />
				</Provider>
			</BrowserRouter>
		);
	});
	it('should render SearchView component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
