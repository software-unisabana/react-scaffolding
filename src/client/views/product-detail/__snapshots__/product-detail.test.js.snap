// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailView should render SearchView component 1`] = `ReactWrapper {}`;
